### PR TITLE
Introducing SmartShop example

### LINKED ISSUE #6: Add SmartShop example

### PR DESCRIPTION
**Pull request by DanielaIvanova**
resolves #6

### CODE REVIEW

**Comment by thepiwo** [comment]
the .forgae-store/ folder should be removed and be included in gitignore

File: README.md [hunk 1]
```diff
@@ -4,3 +4,5 @@
 
 The provided sophia examples or library functions are only tested to the extend represented. None of the code was security audited by aeternity or is provided to be used in production, without thorough checks.
 
+## Libraries
```
Comment (thepiwo) [inline README.md]: this is an example, right?

Comment (DanielaIvanova) [reply README.md]: Shall I make new directory `examples` and move the project there?

Comment (thepiwo) [reply README.md]: yes please :)

**Comment by thepiwo** [comment]
@alekseytsekov could you review this as well?

File: examples/SmartShop/contracts/BuyerContract.aes [hunk 2]
```diff
@@ -0,0 +1,31 @@
+contract Remote =
+  function received_item : () => bool
+  function seller_contract_balance : () => int
+  function check_item_status : () => string
+
+  function check_courier_status : () => string
+  function check_courier_location : () => string
+  function check_courier_timestamp : () => int
```
Comment (alekseytsekov) [inline examples/SmartShop/contracts/BuyerContract.aes]: can you separate these interfaces?

File: examples/SmartShop/contracts/SellerContract.aes [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+    
+  public function check_item_status() =
+    state.status
+
+  private function check_money() : bool =
+    if(Contract.balance >= state.price)
+      true
+    else
+      false
+
```
Comment (alekseytsekov) [inline examples/SmartShop/contracts/SellerContract.aes]: maybe seller should have some collection of products/items? :) ....and have add/remove functionality.

File: examples/SmartShop/contracts/SellerContract.aes [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+    Chain.balance(Call.caller)
+    
+  public function seller_contract_balance() : int =
+    Contract.balance
+    
+  public function check_item_status() =
+    state.status
+
+  private function check_money() : bool =
+    if(Contract.balance >= state.price)
```
Comment (alekseytsekov) [inline examples/SmartShop/contracts/SellerContract.aes]: can you add some kind of map that store every deposit.

File: examples/SmartShop/contracts/BuyerContract.aes [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+  
+contract Buyer =
+  function deposit_to_seller_contract(price : int, key : address) : bool =
+    Chain.spend(key, price)
```
Comment (alekseytsekov) [inline examples/SmartShop/contracts/BuyerContract.aes]: maybe on buyer init should initialize with seller address?

File: examples/SmartShop/contracts/SellerContract.aes [hunk 6]
```diff
@@ -0,0 +1,36 @@
+contract Seller =
+  record state = {seller_key : address, buyer_contract: address, status : string, price : int}
+  
+  public stateful function init(bc : address, price : int) : state =
+    {seller_key = Call.caller, buyer_contract = bc, status = "undefined", price = price }
+    
+  public stateful function send_item() : bool =
+    if(check_money())
```
Comment (alekseytsekov) [inline examples/SmartShop/contracts/SellerContract.aes]: maybe when buyer bought a product/item you should change the owner of product/item

File: examples/SmartShop/contracts/SellerContract.aes (hunk 3 above)
Comment (DanielaIvanova) [reply examples/SmartShop/contracts/SellerContract.aes]: This example project was intended to represent basic implementation and is working with a single product. As an example, we would prefer to keep it simple as it is.

File: examples/SmartShop/contracts/SellerContract.aes (hunk 4 above)
Comment (DanielaIvanova) [reply examples/SmartShop/contracts/SellerContract.aes]: We would prefer to keep it simple as it is.

File: examples/SmartShop/contracts/BuyerContract.aes (hunk 5 above)
Comment (DanielaIvanova) [reply examples/SmartShop/contracts/BuyerContract.aes]: The idea of this example is to show how to make remote calls and very basic seller-buyer behavior implementation.

File: examples/SmartShop/contracts/SellerContract.aes (hunk 6 above)
Comment (DanielaIvanova) [reply examples/SmartShop/contracts/SellerContract.aes]: Same as comment above.

**Pull request merged**